Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/ui/input/AnalyticsUserInputViewModel.kt

```diff
@@ -0,0 +1,177 @@
+    val userInfoItems: LiveData<List<UserInfoItem>> = when (type) {
+        AnalyticsUserInputFragment.InputType.AGEGROUP -> ageGroupSource
+        AnalyticsUserInputFragment.InputType.FEDERALSTATE -> federalStateSource
+        AnalyticsUserInputFragment.InputType.DISTRICT -> districtSource
+    }
+        .catch { Timber.e(it, "Error sourcing list items.") }
+        .asLiveData(context = dispatcherProvider.Default)
+
+    val finishEvent = SingleLiveEvent<Unit>()
+
+    fun selectUserInfoItem(item: UserInfoItem) {
+        when (item.data) {
+            is PpaData.PPAAgeGroup -> {
+                settings.userInfoAgeGroup.update { item.data }
+            }
+            is PpaData.PPAFederalState -> {
+                settings.userInfoFederalState.update { item.data }
+                settings.userInfoDistrict.update { 0 }
+            }
+            is Districts.District -> {
+                settings.userInfoDistrict.update { item.data.districtId }
+            }
+            else -> throw IllegalArgumentException()
+        }
+        finishEvent.postValue(Unit)
+    }
+
+    @StringRes
+    private fun PpaData.PPAFederalState.getStringLabel(): Int = when (this) {
+        PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED -> R.string.analytics_userinput_federalstate_unspecified
+        PpaData.PPAFederalState.FEDERAL_STATE_BW -> R.string.analytics_userinput_federalstate_bw
+        PpaData.PPAFederalState.FEDERAL_STATE_BY -> R.string.analytics_userinput_federalstate_by
+        PpaData.PPAFederalState.FEDERAL_STATE_BE -> R.string.analytics_userinput_federalstate_be
+        PpaData.PPAFederalState.FEDERAL_STATE_BB -> R.string.analytics_userinput_federalstate_bb
+        PpaData.PPAFederalState.FEDERAL_STATE_HB -> R.string.analytics_userinput_federalstate_hb
+        PpaData.PPAFederalState.FEDERAL_STATE_HH -> R.string.analytics_userinput_federalstate_hh
+        PpaData.PPAFederalState.FEDERAL_STATE_HE -> R.string.analytics_userinput_federalstate_he
+        PpaData.PPAFederalState.FEDERAL_STATE_MV -> R.string.analytics_userinput_federalstate_mv
+        PpaData.PPAFederalState.FEDERAL_STATE_NI -> R.string.analytics_userinput_federalstate_ni
+        PpaData.PPAFederalState.FEDERAL_STATE_NRW -> R.string.analytics_userinput_federalstate_nrw
+        PpaData.PPAFederalState.FEDERAL_STATE_RP -> R.string.analytics_userinput_federalstate_rp
+        PpaData.PPAFederalState.FEDERAL_STATE_SL -> R.string.analytics_userinput_federalstate_sl
+        PpaData.PPAFederalState.FEDERAL_STATE_SN -> R.string.analytics_userinput_federalstate_sn
+        PpaData.PPAFederalState.FEDERAL_STATE_ST -> R.string.analytics_userinput_federalstate_st
+        PpaData.PPAFederalState.FEDERAL_STATE_SH -> R.string.analytics_userinput_federalstate_sh
+        PpaData.PPAFederalState.FEDERAL_STATE_TH -> R.string.analytics_userinput_federalstate_th
+        PpaData.PPAFederalState.UNRECOGNIZED -> throw IllegalArgumentException("PpaData.PPAFederalState.UNRECOGNIZED")
```

Comment:
Can we simply this somehow? I see that this enum is coming from the server. In this case I would just create another one for the UI state. We can have more control over it and the fields that are related. what do you think?
This might also simplify https://github.com/corona-warn-app/cwa-app-android/pull/2270/files#diff-8da4ec62b6a184a1706e14e77b4184ee344079fd3a174efdafcf23c4c1c15be0R76-R91

Reply:
Creating another enum or data class would just increase the overhead as we then need to start mapping between additional types. I think this is a very good use-case for extension functions. I have extracted them into an extra file so that  @AlexanderAlferov  can reuse them.